### PR TITLE
Docs: Checkout

### PR DESCRIPTION
**Pull request by Fifciu**
Copy of https://github.com/vuestorefront/vue-storefront/pull/5606 with resolved conflicts

### CODE REVIEW

File: packages/core/docs/commercetools/composables/use-billing.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# `useBilling`
+
+## Features
+
+`useBilling` composable can be use to:
```
Comment (NataliaTepluhina) [inline packages/core/docs/commercetools/composables/use-billing.md:5]:
```suggestion
`useBilling` composable can be used for:
```

File: packages/core/docs/commercetools/composables/use-shipping-provider.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+# `useShippingProvider`
+
+## Features
+
+`useShippingProvider` composable can be use to:
```
Comment (NataliaTepluhina) [inline packages/core/docs/commercetools/composables/use-shipping-provider.md:5]:
```suggestion
`useShippingProvider` composable can be used for:
```

File: packages/core/docs/commercetools/composables/use-shipping-provider.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+* Loading shipping methods for the current cart.
+* Selecting shipping method for the current cart.
+
+## API
+
+- `load` - function for fetching shipping method. When invoked, it requests data from the API and populates `response` key inside `state` property. This method accepts a single optional `params` object. The `params` has the following option:
```
Comment (NataliaTepluhina) [inline packages/core/docs/commercetools/composables/use-shipping-provider.md:12]:
```suggestion
- `load` - function for fetching shipping method. When invoked, it requests data from the API and populates the `response` key inside the `state` property. This method accepts a single optional `params` object. The `params` has the following option:
```

File: packages/core/docs/commercetools/composables/use-shipping.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# `useShipping`
+
+## Features
+
+`useShipping` composable can be use to:
```
Comment (NataliaTepluhina) [inline packages/core/docs/commercetools/composables/use-shipping.md:5]:
```suggestion
`useShipping` composable can be used for:
```

File: packages/core/docs/guide/checkout.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
+# Checkout
+Checkout is a process of providing shipping and billing addresses and selecting shipping and payment methods needed to place an order and pay for it.
+
+## Collecting and saving shipping details
+Shipping details are information about the recipient's address needed to ship the order.
+
+We can load shipping details by calling the `load` method in `useShipping` composable and use the `shipping` property to access it.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:7]:
```suggestion
We can load shipping details by calling the `load` method in `useShipping` composable and accessing the `shipping` property after loading is done.
```

File: packages/core/docs/guide/checkout.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
+# Checkout
+Checkout is a process of providing shipping and billing addresses and selecting shipping and payment methods needed to place an order and pay for it.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:2]: **suggestion**: let's have a consistent newline between headings and text and between text and code snippets
```suggestion
# Checkout

Checkout is a process of providing shipping and billing addresses and selecting shipping and payment methods needed to place an order and pay for it.
```

File: packages/core/docs/guide/checkout.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+
+## Collecting and saving shipping details
+Shipping details are information about the recipient's address needed to ship the order.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:5]:
```suggestion
## Collecting and saving shipping details

Shipping details are information about the recipient's address required to ship the order.
```

File: packages/core/docs/guide/checkout.md [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+    return {
+      shipping
+    };
+  }
+}
+```
+`shipping` property returns `null` if the `load` function was not invoked or nothing is saved.    
+
+We can use `save` method to save shipping details so they're available next time we `load` them.
+```vue{2,15,24}
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:32]:
````suggestion
We can use `save` method to save shipping details so they're available next time we `load` them.

```vue{2,15,24}
````

File: packages/core/docs/guide/checkout.md [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+      shippingForm,
+      save
+    };
+  }
+}
+</script>
+```
+
+## Selecting a shipping method
+`VsfShippingProvider` is a component that aggregates one or more shipping methods from a single provider like FedEx or DHL. This component is usually the only thing that we need to integrate a particular vendor into our project and is always delivered as a third-party integration.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:64]:
```suggestion
## Selecting a shipping method

`VsfShippingProvider` is a component that aggregates one or more shipping methods from a single provider like FedEx or DHL. This component is usually the only thing that we need to integrate a particular vendor into our project and is always delivered as third-party integration.
```

File: packages/core/docs/guide/checkout.md [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+
+The component is responsible for:
+- Loading and displaying available shipping methods.
+- Loading selected shipping method.
+- Selecting and configuring shipping method.
+
+All we have to do is import and put in the view.
+```vue
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:72]:
````suggestion
All we have to do is to import a component and add it to the template.

```vue
````

File: packages/core/docs/guide/checkout.md [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+<VsfShippingProvider
+  @submit="$router.push('/checkout/billing')"
+/>
+```
+
+`VsfShippingProvider` emits `submit` event when shipping method is selected, configured and user clicks submit button.
+
+### Extending `VsfShippingProvider` and reacting to it's events
+We can pass asynchronous functions to the `VsfShippingProvider` component to hook into different events within its lifecycle, override initial function parameters or react to specific events like method selection. We will call these methods "hooks".
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:81]:
```suggestion
### Extending `VsfShippingProvider` and reacting to its events

We can pass asynchronous functions to the `VsfShippingProvider` component to hook into different events within its lifecycle, override initial function parameters or react to specific events like method selection. We will call these methods "hooks".
```

File: packages/core/docs/guide/checkout.md [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+  :afterLoad="afterLoad"
+  :beforeSelect="beforeSelect"
+  :afterSelect="afterSelect"
+  :beforeSelectedDetailsChange="beforeSelectedDetailsChange"
+  :afterSelectedDetailsChange="afterSelectedDetailsChange"
+  :onError="onError"
+/>
+```
+## Collecting and saving billing details
+Billing details are information about the payer's address used by store owners to prepare invoices and payment providers to evaluate the probability of fraud payment.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:105]:
```suggestion
## Collecting and saving billing details

Billing details are information about the payer's address used by store owners to prepare invoices and payment providers to evaluate the probability of fraud payment.
```

File: packages/core/docs/guide/checkout.md [hunk 13]
```diff
@@ -0,0 +1,324 @@
...
+
+We can load billing details by calling the `load` method in `useBilling` composable and use the `billing` property to access it.
+```js{8,16}
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:108]:
````suggestion
We can load billing details by calling the `load` method in `useBilling` composable and accessing the `billing` property after loading is done.

```js{8,16}
````

File: packages/core/docs/guide/checkout.md [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+    return {
+      billing
+    };
+  }
+}
+```
+`billing` property returns `null` if the `load` function was not invoked or nothing is saved.   
+
+We can use `save` method to save billing details.
+```vue{2,15,24}
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:132]:
````suggestion
We can use `save` method to save billing details.

```vue{2,15,24}
````

File: packages/core/docs/guide/checkout.md [hunk 15]
```diff
@@ -0,0 +1,324 @@
...
+      billingForm,
+      save
+    };
+  }
+}
+</script>
+```
+
+## Making an order
+After user provided every information required by our eComerce, we are ready to *make an order*. To do that, we have to call a `make` method from the `useMakeOrder` composable.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:164]:
```suggestion
## Making an order

After the user has provided all the information required by our eCommerce, we are ready to *make an order*. To do that, we have to call a `make` method from the `useMakeOrder` composable.
```

File: packages/core/docs/guide/checkout.md [hunk 16]
```diff
@@ -0,0 +1,324 @@
...
+
+    return {
+      make
+    }
+  }
+}
+```
+
+When the order is created, we can redirect the user to the page thanking them for making an order and refresh the cart.
+```js
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:180]:
````suggestion
When the order is created, we can redirect the user to the page thanking them for making an order and refresh the cart.

```js
````

File: packages/core/docs/guide/checkout.md [hunk 17]
```diff
@@ -0,0 +1,324 @@
...
+      await make();
+      context.root.$router.push(`/checkout/thank-you?order=${order.value.id}`);
+      setCart(null);
+    }
+  }
+}
+```
+
+## Payment providers
+A `VsfPaymentProvider` is a component that provides one or more payment methods. One such component integrates one 3rd party provider of payments like Checkout.com or Adyen. This component is usually the only thing that we need to integrate a particular vendor into our project and is always delivered as a third-party integration.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:198]:
```suggestion
## Payment providers

A `VsfPaymentProvider` is a component that provides one or more payment methods. One such component integrates one third-party provider of payments like Checkout.com or Adyen. This component is usually the only thing that we need to integrate a particular vendor into our project and is always delivered as third-party integration.   
```

File: packages/core/docs/guide/checkout.md [hunk 18]
```diff
@@ -0,0 +1,324 @@
...
+## Payment providers
+A `VsfPaymentProvider` is a component that provides one or more payment methods. One such component integrates one 3rd party provider of payments like Checkout.com or Adyen. This component is usually the only thing that we need to integrate a particular vendor into our project and is always delivered as a third-party integration.   
+
+The component is responsible for:
+- Loading and displaying available payment methods.
+- Loading selected payment method.
+- Picking and configuring payment method.
+
+The first thing we have to do is import and put in the view.
+```vue
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:206]:
````suggestion
The first thing we have to do is to import the component and add it to the template.

```vue
````

File: packages/core/docs/guide/checkout.md [hunk 19]
```diff
@@ -0,0 +1,324 @@
...
+
+export default {
+  components: {
+    VsfPaymentProvider
+  }
+}
+</script>
+```
+
+The next step is making a payment. Each package with a payment provider might use a slightly different approach, but below we described are two most common.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:224]:
```suggestion
The next step is making a payment. Each package with a payment provider might use a slightly different approach, but below we described the two most common.
```

File: packages/core/docs/guide/checkout.md [hunk 20]
```diff
@@ -0,0 +1,324 @@
...
+
+### SDK takes the full control
+If payment provider's SDK handles the whole payment and we can only provide own callbacks for certain events. We want to make an order in `beforePay` async hook.
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:227]:
```suggestion
### SDK takes the full control

If the payment provider's SDK handles the whole payment and we can only provide our own callbacks for certain events. We want to make an order in `beforePay` async hook.
```

File: packages/core/docs/guide/checkout.md [hunk 21]
```diff
@@ -0,0 +1,324 @@
...
+    return {
+      makeOrder
+    }
+  }
+}
+</script>
+```
+
+### SDK allows to externalize pay method
+If payment provider's SDK handles process of configuring payment but allows us to decide when to finalize then:
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:252]:
```suggestion
### SDK allows externalizing pay method

If the payment provider's SDK handles the process of configuring payment but allows us to decide when to finalize then:
```

File: packages/core/docs/guide/checkout.md [hunk 22]
```diff
@@ -0,0 +1,324 @@
...
+      makeOrder,
+      readyToPay
+    };
+  }
+}
+</script>
+```
+
+### Extending `VsfPaymentProvider` and reacting to it's events
+We can pass asynchronous functions to the `VsfPaymentProvider` component to hook into different events within its lifecycle, override initial function parameters or react to specific events like method selection. We will call these methods "hooks".
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:295]:
```suggestion
### Extending `VsfPaymentProvider` and reacting to its events

We can pass asynchronous functions to the `VsfPaymentProvider` component to hook into different events within its lifecycle, override initial function parameters or react to specific events like method selection. We will call these methods "hooks".
```

File: packages/core/docs/guide/checkout.md [hunk 23]
```diff
@@ -0,0 +1,324 @@
...
+  :beforePay="beforePay"
+  :afterPay="afterPay"
+  :beforeSelectedDetailsChange="beforeSelectedDetailsChange"
+  :afterSelectedDetailsChange="afterSelectedDetailsChange"
+  :onError="onError"
+/>
+```
+
+### Why some integrations have a mocked VsfPaymentProvider?
+There are eCommerce backends which do not provide any payment methods out-of-the-box, e.g. commercetools. For these, we provide mock component that has exactly same interface as a real integration so you can easily swap it with a payment integration of your choice. You can find available payment integrations [here](/v2/integrations/)
```
Comment (NataliaTepluhina) [inline packages/core/docs/guide/checkout.md:324]:
```suggestion
### Why some integrations have a mocked `VsfPaymentProvider`?

There are eCommerce backends that do not provide any payment methods out-of-the-box, e.g. commercetools. For these, we provide a mock component that has exactly the same interface as a real integration so you can easily swap it with a payment integration of your choice. You can find available payment integrations [here](/v2/integrations/)
```

File: packages/core/docs/guide/checkout.md [hunk 24]
```diff
@@ -0,0 +1,369 @@
...
+  :afterLoad="afterLoad"
+  :beforeSelect="beforeSelect"
+  :afterSelect="afterSelect"
+  :beforeSelectedDetailsChange="beforeSelectedDetailsChange"
+  :afterSelectedDetailsChange="afterSelectedDetailsChange"
+  :onError="onError"
+/>
+```
+
+:::details Loading and saving a shipping method
```
Comment (filrak) [inline packages/core/docs/guide/checkout.md:111]: how about not making this as toggle and just adding a paragraph

## Accessing shipping method details outside the component

And describing for what typically this composable is used for and that it shares the state with the component and that's it? 

You don't even have to rephrase that much

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**